compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/error.dont-memoize-array-with-capturing-map-after-hook.expect.md

```diff
@@ -0,0 +1,47 @@
+
+## Input
+
+```javascript
+import {useEffect, useState} from 'react';
+import {mutate} from 'shared-runtime';
+
+function Component(props) {
+  const x = [{...props.value}];
+  useEffect(() => {}, []);
+  const onClick = () => {
+    console.log(x.length);
+  };
+  let y;
+  return (
+    <div onClick={onClick}>
+      {x.map(item => {
+        y = item;
+        return <span key={item.id}>{item.text}</span>;
+      })}
+      {mutate(y)}
+    </div>
+  );
+}
+
+export const FIXTURE_ENTRYPOINT = {
+  fn: Component,
+  params: [{value: {id: 0, text: 'Hello!'}}],
+  isComponent: true,
+};
+
+```
+
+
+## Error
+
+```
+  12 |     <div onClick={onClick}>
+  13 |       {x.map(item => {
+> 14 |         y = item;
+     |         ^ InvalidReact: Reassigning a variable after render has completed can cause inconsistent behavior on subsequent renders. Consider using state instead. Variable `y` cannot be reassigned after render (14:14)
+  15 |         return <span key={item.id}>{item.text}</span>;
+  16 |       })}
+  17 |       {mutate(y)}
+```
```

Comment:
hmmm this case is legit since we know `x` is an array and that the .map is executed synchronously. we have an existing notion of functions where the arguments don't escape (noAlias) which is set for Array.prototype.map.

I think we could check that here and allow reassigning functions in the case where noAlias is set on the callee/method. But also...reassigning in a map like this is pretty weird and it's probably fine to disallow it.

Reply:
Good point, and even if it's sketchy might as well be permissive when we already have the tools to do so safely. Updated!